src/devices/src/virtio/net/unixgram.rs

```diff
@@ -19,34 +19,6 @@ pub struct Unixgram {
 impl Unixgram {
     /// Create the backend with a pre-established connection to the userspace network proxy.
     pub fn new(fd: RawFd) -> Self {
-        Self { fd }
-    }
-
-    /// Create the backend opening a connection to the userspace network proxy.
-    pub fn open(path: PathBuf, send_vfkit_magic: bool) -> Result<Self, ConnectError> {
-        let fd = socket(
-            AddressFamily::Unix,
-            SockType::Datagram,
-            SockFlag::empty(),
-            None,
-        )
-        .map_err(ConnectError::CreateSocket)?;
-        let peer_addr = UnixAddr::new(&path).map_err(ConnectError::InvalidAddress)?;
-        let local_addr = UnixAddr::new(&PathBuf::from(format!("{}-krun.sock", path.display())))
-            .map_err(ConnectError::InvalidAddress)?;
-        if let Some(path) = local_addr.path() {
-            _ = unlink(path);
-        }
-        bind(fd, &local_addr).map_err(ConnectError::Binding)?;
-
-        // Connect so we don't need to use the peer address again. This also
-        // allows the server to remove the socket after the connection.
-        connect(fd, &peer_addr).map_err(ConnectError::Binding)?;
-
-        if send_vfkit_magic {
-            send(fd, &VFKIT_MAGIC, MsgFlags::empty()).map_err(ConnectError::SendingMagic)?;
-        }
-
         // macOS forces us to do this here instead of just using SockFlag::SOCK_NONBLOCK above.
```

Comment:
This comment was good in open() since we created the socket above. In new() we need another comment explaining that we handle socket from 2 sources - user socket and socket created in open. The comment in my patch tried to explain this by maybe it should be improved:

```
// For existing socket we must make the socket non-blocking here. When
// opening the socket we can create it non-blocking but not on macOS.
```

Reply:
I've opted for simplifying the comment, since in that context doesn't make sense to talk about macOS.